services/leader/service.go

```diff
@@ -0,0 +1,189 @@
+package leader
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/m3db/m3cluster/services"
+	"github.com/m3db/m3cluster/services/leader/campaign"
+	"github.com/m3db/m3x/errors"
+)
+
+const (
+	leaderKeyPrefix = "_ld"
+	keySeparator    = "/"
+	keyFormat       = "%s/%s"
+)
+
+var (
+	// errClientClosed indicates the election service client has been closed and
+	// no more elections can be started.
+	errClientClosed = errors.New("election client is closed")
+)
+
+type multiClient struct {
+	sync.RWMutex
+
+	closed     uint32
```

Comment:
Probably better to make this a `bool` and use the `RWMutex` to protect its access.